pkg/rehearse/configmaps.go

```diff
@@ -32,6 +33,53 @@ const (
 	rehearseLabelPull = "ci.openshift.org/rehearse-pull"
 )
 
+// ConfigMaps holds the data about the ConfigMaps affected by a rehearse run
+type ConfigMaps struct {
+	// Paths is a set of repo paths that changed content and belong to some ConfigMap
+	Paths sets.String
+	// Names is a mapping from production ConfigMap names to rehearse-specific ones
+	Names map[string]string
+	// ProductionNames is a set of production ConfigMap names
+	ProductionNames sets.String
+	// Patterns is the set of config-updater patterns that cover at least one changed file
+	Patterns sets.String
+}
+
+// NewConfigMaps populates a ConfigMaps instance
+func NewConfigMaps(paths []string, purpose string, prNumber int, configUpdaterCfg prowplugins.ConfigUpdater) (ConfigMaps, error) {
+	cms := ConfigMaps{
+		Paths:           sets.NewString(paths...),
+		Names:           map[string]string{},
+		ProductionNames: sets.NewString(),
+		Patterns:        sets.NewString(),
+	}
+
+	var errs []error
+	for _, path := range paths {
+		cmName, pattern, err := config.ConfigMapName(path, configUpdaterCfg)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		cms.Names[cmName] = tempConfigMapName(purpose, cmName, prNumber)
+		cms.ProductionNames.Insert(cmName)
+		cms.Patterns.Insert(pattern)
+	}
+
+	return cms, kutilerrors.NewAggregate(errs)
+}
+
+func tempConfigMapName(purpose, source string, prNumber int) string {
+	// Object names can't be too long so we truncate the hash. This increases
+	// chances of collision but we can tolerate it as our input space is tiny.
+	pr := strconv.Itoa(prNumber)
+	maxLen := 253 - len("rehearse---") - len(purpose) - len(pr)
+	if len(source) > maxLen {
+		source = source[:maxLen]
+	}
+	return fmt.Sprintf("rehearse-%s-%s-%s", pr, purpose, source)
+}
```

Comment:
@bbguimaraes I wanted to consult the naming of the temporary CMs with you here. I remember we initially put PR numbers into the name, and then you changed it to contain SHAs instead. I could not keep the SHAs because in my model I disconnect the CM name from its content (because I now work purely with CM names). 

Was there any problem with PR number-based names? If so I would consider reintroducing the effect of the content on the name (would need to sum multiple hashes though...), or even using the `BUILD_ID` or somehting like that.

Reply:
Here's a blast from the past:

The only change I made was to use the hash already present in Git instead of recalculating it by other means: https://github.com/openshift/ci-operator-prowgen/pull/167.

Re PR vs. hash: https://github.com/openshift/ci-operator-prowgen/pull/100#pullrequestreview-210698890.